infrastructure/src/test/java/org/corfudb/infrastructure/LogUnitServerCacheTest.java

```diff
@@ -0,0 +1,77 @@
+package org.corfudb.infrastructure;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import org.corfudb.common.metrics.micrometer.MeterRegistryProvider.MeterRegistryInitializer;
+import org.corfudb.infrastructure.BatchProcessor.BatchProcessorContext;
+import org.corfudb.infrastructure.LogUnitServer.LogUnitServerConfig;
+import org.corfudb.infrastructure.log.InMemoryStreamLog;
+import org.corfudb.infrastructure.log.StreamLog;
+import org.corfudb.protocols.wireprotocol.DataType;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.util.LambdaUtils.BiOptional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+class LogUnitServerCacheTest {
+    private final Random rnd = new Random();
+
+    @Test
+    public void testMetrics() {
```

Comment:
Can we add a test without any accesses to exercise the `NaN` case?

Reply:
done